### PR TITLE
Proof of concept of a smoke test runner

### PR DESCRIPTION
**Pull request by stevenwilkin**
The bare minimum for the time being:
- uses FakeRedis
- no email alerts of failed tests
- test runner webhooks are unprotected
- no history of test runs
- each environment can only have 1 test script

Should be enough for some conversation though @andrewgarner @gjvis

### CODE REVIEW

**Comment by gjvis** [comment]
Looks like a good starting point. Lots of opportunity to extend and improve once we know more about how we want to hook this in.

:+1: 

@andrewgarner ?

File: domain/environment.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+  end
+
+  def script
+    self.class.script
+  end
+
+  def check
+    output = run_script(script)
+    self.class.redis.set("#{self.class.to_s.downcase}_status", exit_status)
```
Comment (andrewgarner) [inline domain/environment.rb]: How about using a namespace to DRY this up?

``` rb
def redis
  @redis ||= Redis::Namespace.new(self.class.to_s.downcase, redis: Redis.new)
end

# then

self.class.redis.set(:status, exit_status)
```

**Comment by andrewgarner** [comment]
:+1: 

Maybe a JSON view would be handy too so that we can easily access the statuses for things like radiating?

File: domain/environment.rb (hunk 1 above)
Comment (stevenwilkin) [inline domain/environment.rb]: I've checked and `Redis::Namespace` plays along with `FakeRedis` ok.

I originally thought I needed to maintain just a single `Redis` instance for the sake of `FakeRedis` and the tests and so I stored it as a class variable on `Environment` but this doesn't seem to be necessary so creating a namespaced connection for each subclass should work fine.

**Comment by stevenwilkin** [comment]
A JSON endpoint sounds like a good idea @andrewgarner, it would mean we could display the status for each environment as either "Up" or "Down" on Geckoboard using [their monitoring widget](https://developer.geckoboard.com/polling-overview#monitoring).

Perhaps using that type of widget would allow us to bypass having to write code to fire off emails on test failure?

**Pull request merged**